tests/test_dragon_installer.py

```diff
@@ -0,0 +1,519 @@
+    extract_path.mkdir()
+
+    # verify archive doesn't exist & folder does
+    assert not archive_path.exists()
+    assert extract_path.exists()
+
+    cleanup(archive_path, extract_path)
+
+    # verify folder is gone after cleanup
+    assert not archive_path.exists()
+    assert not extract_path.exists()
+
+
+def test_cleanup_nonempty_extraction_directory(test_dir: str) -> None:
+    """Ensure that the cleanup method works when the extraction directory
+    is NOT empty"""
+    archive_path = pathlib.Path(test_dir) / mock_archive_name
+    extract_path = pathlib.Path(test_dir) / mock_archive_name.replace(".tar.gz", "")
+    extract_path.mkdir()
+
+    num_files = 10
+    for i in range(num_files):
+        content = extract_path / f"{i:04}.txt"
+        content.write_text(f"i am file {i}\n")
+
+    files = list(extract_path.rglob("*.txt"))
+
+    assert len(files) == num_files
+
+    cleanup(archive_path, extract_path)
+
+    # verify folder is gone after cleanup
+    assert not archive_path.exists()
+    assert not extract_path.exists()
+
+
+def test_cleanup_no_extract_path(test_dir: str) -> None:
+    """Ensure that the cleanup method doesn't bomb when called with
+    missing extraction directory; simulate failed extract"""
+    archive_path = pathlib.Path(test_dir) / mock_archive_name
+    extract_path = pathlib.Path(test_dir) / mock_archive_name.replace(".tar.gz", "")
+
+    # create an archive to clean up
+    with tarfile.TarFile.open(archive_path, mode="w:gz") as tar:
+        tar.add(__file__)  # add current file to avoid empty tar
+
+    # verify archive exists before cleanup
+    assert archive_path.exists()
+
+    cleanup(archive_path, extract_path)
+
+    # verify archive is gone after cleanup
+    assert not archive_path.exists()
+    assert not extract_path.exists()
+
+
+def test_cleanup_no_archive(test_dir: str) -> None:
+    """Ensure that the cleanup method doesn't bomb when called with
+    missing archive"""
+    archive_path = pathlib.Path(test_dir) / mock_archive_name
+    extract_path = pathlib.Path(test_dir) / mock_archive_name.replace(".tar.gz", "")
+
+    extract_path.mkdir()
+
+    # verify archive exists before cleanup
+    assert extract_path.exists()
+
+    cleanup(archive_path, extract_path)
+
+    # verify archive is gone after cleanup
+    assert not archive_path.exists()
+    assert not extract_path.exists()
+
+
+def test_expand_archive(test_dir: str) -> None:
+    """Verify archive is expanded into expected location w/correct content"""
+    archive_path = pathlib.Path(test_dir) / mock_archive_name
+    exp_path = pathlib.Path(test_dir) / mock_archive_name.replace(".tar.gz", "")
+    num_files = 10
+
+    # create an archive to clean up
+    with tarfile.TarFile.open(archive_path, mode="w:gz") as tar:
+        for i in range(num_files):
+            content = pathlib.Path(test_dir) / f"{i:04}.txt"
+            content.write_text(f"i am file {i}\n")
+            tar.add(content)
```

Comment:
Similar note here: It looks like this tarfile is remade in multiple tests. Could it be promoted to a fixture?